src/Common/EnumerableExtensions.cs

```diff
@@ -16,7 +16,25 @@ internal static class EnumerableExtensions
     [SuppressMessage("Performance", "ECS0900:Minimize boxing and unboxing", Justification = "Should revisit. Suppressing for now to unblock refactor.")]
     public static TSource? DefaultIfNotSingle<TSource>(this ImmutableArray<TSource> source, Func<TSource, bool> predicate)
     {
-        return source.AsEnumerable().DefaultIfNotSingle(predicate);
+        bool isFound = false;
```

Comment:
This is duplicative. Evaluate with a performance benchmark and provide quantitative evidence why this needs to be changed. If the change is substantial, write a comment indicating the findings so that a future maintainer does not revert